evm/spec/tables/cpu.tex

```diff
@@ -1,4 +1,73 @@
 \subsection{CPU}
 \label{cpu}
 
-TODO
+The CPU is the central component of the zkEVM. Like any CPU, it reads instructions, executes them and modifies the state (its registers and the memory)
+accordingly. The constraining of some complex instructions (e.g. Keccak hashing) is delegated to the other tables.
+This section will only briefly present the CPU and its columns. Details about the CPU logic will be provided later.
+
+\subsubsection{CPU flow}
+
+An execution run is made of three distinct parts:
+\begin{itemize}
+    \item \textbf{Bootstrapping:} The CPU starts by writing all the kernel code in memory, and hashing it. The hash is then compared to a public value shared with
+the verifier to ensure that the kernel code is correct.
+    \item \textbf{CPU cycles:} The bulk of the execution. In each row, the CPU reads the current code at the program counter (PC) address, and executes it. The current code can be the kernel code,
+or whichever code is being executed in the current context (transaction code or contract code). Executing an instruction consists in modifying the registers, possibly
+performing some memory operations, and updating the PC.
+\item \textbf{Padding:} At the end of the execution, we need to pad the length to the next power of two. Halting happens when the PC reaches the special halting label
+in the kernel. Constraints ensure that every subsequent row is a padding row and that you can't resume execution.
+\end{itemize}
+
+\subsubsection{CPU columns}
+
+\paragraph*{Registers:} \begin{itemize}
+    \item \texttt{is\_bootstrap\_kernel}: Boolean indicating whether this is a bootstrapping row or not. It must be 1 at the first row, then switch to 0 until the end.
+    \item \texttt{context}: Indicates which context we are in. 0 for the kernel, and a positive integer for every user context. Incremented by 1 at every call.
```

Comment:
Is the context stateful? If I execute a kernel call mid-program then `context` is set to `0` but when I come back to userland execution, is it restored to be a positive integer or does it restart from `1`?

Reply:
When you execute a syscall, `context` doesn't change but `is_kernel_mode` is set to `1` and `code_context` is set to `0`.
You're only in `context` `0` before executing the transaction (to check the validity of the tries before, to parse the transaction etc.) and after the transaction (to check the validity of the trees after, to update the receipt trie, to halt...). I'll add something about it.

Reply:
Aha, ok, that clears it up. Yes, do please add this info! :)